docs/e2e/extension-e2e-guidelines.md

```diff
@@ -314,3 +315,48 @@ Test Name: Connects to a Hardware wallet for Trezor
 ```
 
 ✅ Proposed solution: The Trezor import flow involves opening the Trezor website, then the user takes additional steps on that website to connect the device. A fake version of this website could be created for testing, and the test build could be updated to use this fake version. It's also worth investigating a phishing detection solution, such as replacing Github.com with an empty page.
+
+## Page Object Model (POM)
+
+POM, or Page Object Model, is a design pattern commonly utilized for writing automated end-to-end tests. A page object is a instance of a class that acts as an interface for the page of the application under test.
+This design pattern is being adopted due to its numerous benefits, including improved test maintenance, increased code reusability, and enhanced readability of test scripts. By abstracting the UI details into page objects, changes to the application's UI require updates only in the page objects, significantly reducing the effort needed to maintain tests as the application evolves. The adoption of POM in our e2e testing strategy aims to leverage these advantages to create a more robust, maintainable, and efficient testing framework.
+
+### Composition of a Page Class
+
+Each page class is composed of:
+
+- **Locators**: HTML elements.
+- **Action Methods**: Methods for interacting with the elements.
+- **Check Methods**: Methods that assert the status of elements.
+
+A test creates page objects and interacts with web elements by calling methods of those page objects.
+
+### Best Practices
+
+- A page object should represent meaningful elements of a page and not necessarily a complete page. It can represent a component of a page, like a navbar. See example [here.](https://github.com/MetaMask/contributor-docs/examples/extension-e2e-page-object-model/homepage.ts#L6)
+- All locators should be kept in the page object file. Carefully define locators for elements in page objects, opting for robust locators since they will be extensively used across various locations. For guidance on crafting these resilient locators, refer to the [element locators section](#element-locators), which outlines the recommended strategies for their creation and usage.
+- Page objects should remain independent and not invoke other page objects to prevent circular references, ensuring they are typically isolated from each other. For handling complex workflows that require interaction across multiple pages, _processes_ should be implemented. Check out the [implementation here.](https://github.com/MetaMask/contributor-docs/examples/extension-e2e-page-object-model/login.process.ts#L14) This approach enables the incorporation of all relevant page objects to support specific flows, such as login, sending a transaction, or creating a swap. A dedicated `processes` folder is used to organize and manage these complex workflows.
```

Comment:
I don't know if the concept of processes was taken from somewhere else, but I wonder if there is a different name we ought to use here. I feel that the name "processes" is a bit overloaded; it's usually used to refer either to a program (from a technical perspective) or a routine (from a non-technical perspective). In a way I guess a process _is_ a routine, but since they represent user flows I'm curious if they should be called "flows" instead?

Reply:
Yeah, I agree that the term "flows" is clearer and more intuitive. I'll make the update to use "flows" instead of "processes" in this PR. Additionally, after this PR is merged, I plan to consistently use "flow" in all related  POM work to maintain clarity and coherence.